apps/meteor/ee/server/apps/communication/rest.ts

```diff
@@ -1124,14 +1124,42 @@ export class AppsRestApi {
 						return API.v1.failure('Invalid status provided, it must be "status" field and a string.');
 					}
 
-					const prl = manager.getOneById(this.urlParams.id);
+					const baseUrl = orchestrator.getMarketplaceUrl();
+					const { id: appId } = this.urlParams;
+					const { version, status } = this.bodyParams;
+
+					const headers = getDefaultHeaders();
+
+					const mktAppsUrl = new URL(`${baseUrl}/v1/apps/${appId}`);
+
+					mktAppsUrl.searchParams.set('appVersion', String(version));
+
+					const request = await fetch(mktAppsUrl.toString(), {
+						headers,
+					});
```

Comment:
Technically, we have the `response` assigned to the variable here, not the `request` :stuck_out_tongue: But what I want to point out here is: you're making the request _first thing when the endpoint is hit_ - however we just want to prevent apps from _being enabled_ if they are enterprise only and the workspace doesn´t have a valid license anymore.

You could prevent unnecessary network calls if you just did this in the cases where it is required